lib/puppet/provider/foreman_resource/rest_v3.rb

```diff
@@ -70,7 +70,11 @@ def request(method, path, params = {}, data = nil, headers = {})
     attempts = 0
     begin
       debug("Making #{method} request to #{uri}")
-      response = oauth_consumer.request(method, uri.to_s, generate_token, {}, data, headers)
+      if [:post, :put, :patch].include?(method)
+        response = oauth_consumer.request(method, uri.to_s, generate_token, {}, data, headers)
+      else
+        response = oauth_consumer.request(method, uri.to_s, generate_token, {}, headers)
```

Comment:
Is GET the only one this applies to? I might flip the conditions around in that case.  Could `:delete` for example be a possible method?